packages/react-client/src/ReactFlightClient.js

```diff
@@ -1808,47 +1848,57 @@ function loadServerReference<A: Iterable<any>, T>(
   return (null: any);
 }
 
+function isReactElementOrArrayLike(
+  value: any,
+  // eslint-disable-next-line no-undef
+): value is {_debugInfo: null | ReactDebugInfo, ...} {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    (isArray(value) ||
+      typeof value[ASYNC_ITERATOR] === 'function' ||
+      value.$$typeof === REACT_ELEMENT_TYPE)
+  );
+}
+
+function isLazy(
+  value: any,
+  // eslint-disable-next-line no-undef
+): implies value is LazyComponent<
+  React$Element<any>,
+  SomeChunk<React$Element<any>>,
+> {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    value.$$typeof === REACT_LAZY_TYPE
+  );
+}
```

Comment:
There can be performance benefits to having these be inline because when the type / hidden class check is done in the VM it can share that check with other code around the same place instead of having to check it again for the next member access inside that branch.


It's not always better depending on how unusual the code path is, but that's why I favor always having these inline in most cases.